nav2_mppi_controller/CMakeLists.txt

```diff
@@ -65,7 +65,7 @@ if(COMPILER_SUPPORTS_FMA)
 endif()
 
 # If building one the same hardware to be deployed on, try `-march=native`!
-add_compile_options(-O3 -finline-limit=10000000 -ffp-contract=fast -ffast-math -mtune=generic)
```

Comment:
Readd

Reply:
This causes: 

```
/usr/include/c++/13/bits/stl_algobase.h:437:30: error: ‘void* __builtin_memcpy(void*, const void*, long unsigned int)’ forming offset 8 is out of the bounds [0, 8] of object ‘shape’ with type ‘index_type’ {aka ‘std::array<long unsigned int, 1>’} [-Werror=array-bounds=]
  437 |             __builtin_memmove(__result, __first, sizeof(_Tp) * _Num);
```

Ideally the compile options that are set in nav2_package() should not apply to external packages. Reading around it looks like this is possible but I'm at my limit of Cmake knowledge to understand how to make this happen

Reply:
`target_compile_options` perhaps for that one then?

Reply:
target_link_libraries provides a way  to suppress warnings from third party headers SYSTEM keyword https://stackoverflow.com/a/70169343 

ament_target_dependencies also provides this option.
https://github.com/ament/ament_cmake/blob/fdbf4574d7ccc67c29d63f906ddbd88017eb9ecc/ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake#L51

Introduced 4 years ago so might do the job you want.
https://github.com/ament/ament_cmake/commit/10a6ceb471a717113289e2486eec513db26e85d6